PhysicsTools/Heppy/interface/MuScleFitCorrector.h

```diff
@@ -0,0 +1,312 @@
+#define PhysicsTools_Heppy_MuScleFitCorrector_h
+
+
+/**
+ * MuScleFitCorrector class
+ * Author M. De Mattia - 18/11/2008
+ * Author S. Casasso   - 25/10/2012
+ * Author E. Migliore  - 25/10/2012
+ */
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include "PhysicsTools/Heppy/interface/MuScleFitCorrector_Functions.h"
+#include "TLorentzVector.h"
+#include "TRandom3.h"
+
+
+/**
+ * This is used to have a common set of functions for the specialized templates to use.
+ * The constructor receives the name identifying the parameters for the correction function.
+ * It reads the parameters from a txt file in data/.
+ */
+
+namespace heppy {
+
+class MuScleFitCorrector
+{
+ public:
+  /**
+   * The constructor takes a string identifying the parameters to read. It
+   * parses the txt file containing the parameters, extracts the index of the
+   * correction function and saves the corresponding pointer. It then fills the
+   * vector of parameters.
+   */
+  MuScleFitCorrector(const TString& identifier)
+    {
+      fileName_ = identifier;
+      readParameters( identifier ); 
+      gRandom_ = new TRandom3();
+    }
+  // dummy empty constructor to allow dictionaries
+  MuScleFitCorrector() {}  
+  ~MuScleFitCorrector() {}
+  
+  // Returns a pointer to the selected function
+  scaleFunctionBase<double * > * getScaleFunction() { return scaleFunction_; }
+  resolutionFunctionBase<double * > * getResolFunction() { return resolutionFunction_; }
+  
+  // Method to get correct pt
+  double getCorrectPt(const TLorentzVector &lorentzVector, const int & chg){
+    
+    // Loop on all the functions and apply them iteratively on the pt corrected by the previous function.
+    double pt = lorentzVector.Pt();
+    pt = ( scaleFunction_->scale( pt, lorentzVector.Eta(), lorentzVector.Phi(), chg, scaleParArray_) );
+    return pt;
+  };
+  
+  // Return the squared difference of relative pT resolutions data-MC
+  double getSigmaPtDiffSquared(const double & pt, const double & eta){
+  
+    double sigmaPtData = resolutionFunction_->sigmaPt(pt, eta, resolDataParArray_);
+    double sigmaPtMC = resolutionFunction_->sigmaPt(pt, eta, resolMCParArray_);
+    return (sigmaPtData*sigmaPtData - sigmaPtMC*sigmaPtMC);
+    
+  };
+
+  
+  // Method to get correct pt (baseline)
+  double getSmearedPt(const TLorentzVector & lorentzVector, const int & chg, bool fake){
+    
+    double pt = lorentzVector.Pt();
+    double eta = lorentzVector.Eta();
+    double squaredDiff = getSigmaPtDiffSquared(pt,eta);
+    if (squaredDiff < 0) return pt;
+    double Cfact = 0.;
+    if (fileName_.Contains("smearPrompt")) Cfact = 0.85; // smear Summer12 against 2012 Prompt data
+    else if (fileName_.Contains("smearReReco")) Cfact = 0.75; // smear Summer12 against 2012 ReReco data
+    else if (fileName_.Contains("2011_MC")) Cfact = 0.8; // smear Fall11 against 2011 ReReco data
+    else {
+      std::cout<<"Are you sure you want to smear data??"<<std::endl;
```

Comment:
Please do not use std::cout, edm::LogInfo or edm::LogWarning are the supported methods.